### DISCUSSION TITLE
Very confused about module binaries

### DISCUSSION THREAD

**Discussion by davidmasp** [nextflow-io/nextflow#5203] [Q&A] [answered]
Hi,

I am trying to include module binaries on my pipeline. Essentially I have a "nextflow module" in a directory with a main.nf script that includes my processes. In addition to that I have a folder in the same directory as `resources/usr/bin` where I placed a python script with exec. permissions (see below).

```
nf_base/
├── main.nf
└── resources
    └── usr
        └── bin
            └── plot.py

3 directories, 2 files
```

From here I try to follow [the documentation](https://www.nextflow.io/docs/latest/module.html#module-binaries) and include `nextflow.enable.moduleBinaries = true` in the pipeline. Then I get the error:

`ERROR ~ No such variable: moduleBinaries`

The same happens both when I include the statement in the pipeline or in the module file. What I am doing wrong?

My nextflow version is: `nextflow version 24.04.2.5914`

When I remove the statement, then the `.command.run` file seems to be only loading the PATH for the pipeline directory and not the module bin directory. Am I missing something?

**Comment by pditommaso** [comment, marked as answer]
This looks like a bug. If you but in the `nextflow.config` should work. Willing to open an issue?

**Reply by davidmasp** [reply]
Sure, I ll try generate something easy to reproduce. I have included this in my config file and works as expected, thanks for the tip.

**Reply by davidmasp** [reply]
Done, see here

https://github.com/nextflow-io/nextflow/issues/5204
